queries/idris/highlights.scm

```diff
@@ -0,0 +1,208 @@
+; ------------------------------------------------------------------------------
+; Literals and comments
+(integer) @number
+
+(literal
+  (number)) @number.float
+
+(char) @character
+
+[
+  (string)
+  (pat_string)
+  (triple_quote_string)
+] @string
+
+(comment) @comment
+
+; ------------------------------------------------------------------------------
+; Punctuation
+[
+  "("
+  ")"
+  "{"
+  "@{"
+  "}"
+  "["
+  "[<"
+  "]"
+] @punctuation.bracket
+
+[
+  (comma)
+  (colon)
+  (pat_op)
+  (tuple_operator)
+] @punctuation.delimiter
+
+(pat_name
+  (loname) @variable.parameter)
+
+; ------------------------------------------------------------------------------
+; Types
+(signature
+  (loname) @type)
+
+(signature
+  (caname) @constructor)
+
+(caname) @constructor
+
+; ------------------------------------------------------------------------------
+; Keywords, operators, imports
+[
+  "if"
+  "then"
+  "else"
+  "case"
+  "of"
+] @keyword.conditional
+
+(module) @module
+
+[
+  "import"
+  "module"
+  "namespace"
+  "parameters"
+] @keyword.import
+
+[
+  (operator)
+  (equal)
+  (wildcard)
+  "."
+  "|"
+  "=>"
+  "⇒"
+  "<="
+  "⇐"
+  "->"
+  "→"
+  "<-"
+  "←"
+  "\\"
+  "`"
+] @operator
+
+(qualified_loname
+  (caname) @module)
+
+(qualified_caname
+  (caname) @constructor)
```

Comment:
redundant with the above `(caname) @constructor`

Reply:
Also, you need to be careful about the pattern order here: general captures need to be first, then overwritten later by more specific captures!